src/MTopJetPreSelectionModule3Jets.cxx

```diff
@@ -0,0 +1,204 @@
+#include <iostream>
+#include <memory>
+
+#include <UHH2/core/include/AnalysisModule.h>
+#include <UHH2/core/include/Event.h>
+#include <UHH2/core/include/Selection.h>
+
+#include <UHH2/common/include/NSelections.h>
+#include <UHH2/common/include/LumiSelection.h>
+#include <UHH2/common/include/TriggerSelection.h>
+#include <UHH2/common/include/JetCorrections.h>
+#include <UHH2/common/include/ObjectIdUtils.h>
+#include <UHH2/common/include/JetIds.h>
+#include <UHH2/common/include/TopJetIds.h>
+#include <UHH2/common/include/TTbarGen.h>
+#include <UHH2/common/include/Utils.h>
+#include <UHH2/common/include/AdditionalSelections.h>
+#include "UHH2/common/include/YearRunSwitchers.h"
+
+// Hists
+#include <UHH2/common/include/ElectronHists.h>
+#include <UHH2/common/include/EventHists.h>
+#include <UHH2/common/include/MuonHists.h>
+#include <UHH2/common/include/JetHists.h>
+#include <UHH2/common/include/TTbarGenHists.h>
+#include <UHH2/MTopJet_UL/include/MTopJetHists.h>
+//
+#include <UHH2/MTopJet_UL/include/ModuleBASE.h>
+#include <UHH2/MTopJet_UL/include/RecoSelections.h>
+#include <UHH2/MTopJet_UL/include/GenSelections.h>
+
+using namespace std;
+
+class MTopJetPreSelectionModule3Jets : public ModuleBASE {
+
+public:
+  explicit MTopJetPreSelectionModule3Jets(uhh2::Context&);
+  virtual bool process(uhh2::Event&) override;
+
+protected:
+
+  // selections
+  std::unique_ptr<uhh2::Selection> lumi_sel;
+
+  std::unique_ptr<uhh2::Selection> genmttbar_sel;
+  std::unique_ptr<uhh2::Selection> met_sel;
+  std::unique_ptr<uhh2::Selection> muon_sel;
+  std::unique_ptr<uhh2::Selection> elec_sel;
+  std::unique_ptr<uhh2::Selection> SemiLepDecay;
+  std::unique_ptr<uhh2::Selection> GenMuonPT;
+  std::unique_ptr<uhh2::Selection> GenElecPT;
+
+  std::unique_ptr<uhh2::AnalysisModule> ttgenprod;
+
+  // handles for output
+  Event::Handle<bool>h_recsel;
+  Event::Handle<bool>h_gensel;
+  Event::Handle<std::vector<TopJet>>h_fatjets;
+
+  // bools
+  bool isMC;
+  bool debug;
+  
+};
+
+/*
+███    ███  ██████  ██████  ██    ██ ██      ███████
+████  ████ ██    ██ ██   ██ ██    ██ ██      ██
+██ ████ ██ ██    ██ ██   ██ ██    ██ ██      █████
+██  ██  ██ ██    ██ ██   ██ ██    ██ ██      ██
+██      ██  ██████  ██████   ██████  ███████ ███████
+*/
+
+MTopJetPreSelectionModule3Jets::MTopJetPreSelectionModule3Jets(uhh2::Context& ctx){
+
+  debug = string2bool(ctx.get("Debug","false")); // look for Debug, expect false if not found
+
+  //// CONFIGURATION
```

Comment:
In debug statements either before the code (prefered) or afterwards but not with `End`.
```suggestion
  //// CONFIGURATION
  if(debug) cout << "Setup configurations" << endl;
```